src/Trace/Trace.php

```diff
@@ -0,0 +1,195 @@
+
+    /**
+     * @var string The trace id for this trace. 128-bit numeric formatted as a 32-byte hex string
+     */
+    private $traceId;
+
+    /**
+     * @var TraceSpan[] List of TraceSpans to report
+     */
+    private $spans = [];
+
+    /**
+     * Instantiate a new Trace instance.
+     *
+     * @param ConnectionInterface $connection The connection to Stackdriver Trace.
+     * @param string $projectId The id of the project this trace belongs to.
+     * @param string $traceId [optional] The id of the trace. If not provided, one will be generated
+     *        automatically for you.
+     * @param array $spans [optional] Array of TraceSpan constructor arguments. See
+     *        {@see Google\Cloud\Trace\TraceSpan::__construct()} for configuration details.
+     * }
+     */
+    public function __construct(ConnectionInterface $connection, $projectId, $traceId = null, $spans = null)
+    {
+        $this->connection = $connection;
+        $this->projectId = $projectId;
+        $this->traceId = $traceId ?: $this->generateTraceId();
+        if ($spans) {
+            $this->spans = array_map(function ($span) {
+                return new TraceSpan($span);
+            }, $spans);
+        }
+    }
+
+    /**
+     * Retrieves the trace's id.
+     *
+     * @return string
+     */
+    public function traceId()
+    {
+        return $this->traceId;
+    }
+
+    /**
+     * Returns a serializable array representing this trace. If no span data
+     * is cached, a network request will be made to retrieve it.
+     *
+     * @see https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects.traces/get Traces get API documentation.
+     *
+     * @param array $options [optional] Configuration Options
+     * @return array
+     */
+    public function info(array $options = [])
+    {
+        // We don't want to maintain both an info array and array of TraceSpans,
+        // so we'll rely on the presence of the loaded/specified spans for whether
+        // or not we should fetch remote data.
+        if (!$this->spans) {
+            $this->reload($options);
+        }
+
+        return [
+            'projectId' => $this->projectId,
+            'traceId' => $this->traceId,
+            'spans' => array_map(function ($span) {
+                return $span->info();
+            }, $this->spans)
+        ];
+    }
+
+    /**
+     * Triggers a network request to load a span's details.
+     *
+     * @see https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects.traces/get Traces get API documentation.
+     *
+     * @param array $options [optional] Configuration Options
+     */
+    public function reload(array $options = [])
+    {
+        $trace = $this->connection->getTrace([
+            'projectId' => $this->projectId,
+            'traceId' => $this->traceId
+        ] + $options);
+
+        if (empty($trace)) {
+            throw new NotFoundException('Trace ID does not exist', 404);
```

Comment:
Note: looks like `traces.get` will return a 404 on missing in the future. We can remove this check once that goes live (but functionality will be identical except for the error message string).